adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/SliceRDD.scala

```diff
@@ -0,0 +1,232 @@
+    SliceArray(arr, maxWidth)
+  }
+}
+
+private[adam] class SliceArraySerializer extends IntervalArraySerializer[ReferenceRegion, Slice, SliceArray] {
+
+  protected val kSerializer = new ReferenceRegionSerializer
+  protected val tSerializer = new AvroSerializer[Slice]
+
+  protected def builder(arr: Array[(ReferenceRegion, Slice)],
+                        maxIntervalWidth: Long): SliceArray = {
+    SliceArray(arr, maxIntervalWidth)
+  }
+}
+
+object SliceRDD {
+
+  /**
+   * Helper function for building a SliceRDD when no
+   * sequence dictionary is available.
+   *
+   * @param rdd Underlying RDD. We recompute the sequence dictionary from
+   *   this RDD.
+   * @return Returns a new SliceRDD.
+   */
+  def apply(rdd: RDD[Slice]): SliceRDD = {
+
+    // get sequence dictionary
+    val sd = new SequenceDictionary(rdd.flatMap(slice => {
+      if (slice.getName != null && slice.getLength != null) {
+        Some(SequenceRecord(slice.getName, slice.getLength))
+      } else {
+        None
+      }
+    }).distinct
+      .collect
+      .toVector)
+
+    SliceRDD(rdd, sd)
+  }
+}
+
+/**
+ * A GenomicRDD that wraps Slice data.
+ *
+ * @param rdd An RDD of genomic Slices.
+ * @param sequences The reference genome this data is aligned to.
+ */
+case class SliceRDD(rdd: RDD[Slice],
+                    sequences: SequenceDictionary) extends AvroGenomicRDD[Slice, SliceRDD] with Logging {
+
+  protected def buildTree(rdd: RDD[(ReferenceRegion, Slice)])(
+    implicit tTag: ClassTag[Slice]): IntervalArray[ReferenceRegion, Slice] = {
+    IntervalArray(rdd, SliceArray.apply(_, _))
+  }
+
+  /**
+   * Save slices as Parquet or FASTA.
+   *
+   * If filename ends in .fa or .fasta, saves as FASTA. If not, saves slices
+   * to Parquet. Defaults to 60 character line length, if saving to FASTA.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   */
+  def save(filePath: java.lang.String, asSingleFile: java.lang.Boolean) {
+    if (filePath.endsWith(".fa") || filePath.endsWith(".fasta")) {
+      saveAsFasta(filePath, asSingleFile = asSingleFile)
+    } else {
+      if (asSingleFile) {
+        log.warn("asSingleFile = true ignored when saving as Parquet.")
+      }
+      saveAsParquet(new JavaSaveArgs(filePath))
+    }
+  }
+
+  /**
+   * Save slices in FASTA format.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   * @param lineWidth Hard wrap FASTA formatted slice at line width, default 60.
+   */
+  def saveAsFasta(filePath: String, asSingleFile: Boolean = false, lineWidth: Int = 60) {
+
+    def toFasta(slice: Slice): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(slice.getName)
+      Option(slice.getDescription).foreach(n => sb.append(" ").append(n))
+      sb.append(s" slice.getStart-slice.getEnd:slice.getStrand")
```

Comment:
OOC, does this syntax come from somewhere? If so, I think it'd be good to document it inline/in the Scaladoc.